pyemma/coordinates/data/util/frames_from_file.py

```diff
@@ -103,7 +103,10 @@ def frames_from_files(files, top, frames, chunksize=1000, stride=1, verbose=Fals
         largest_ind_in_traj = np.max(inds_by_traj)
         if isinstance(reader, FeatureReader):
             length = reader.trajectory_length(itraj)
+        elif isinstance(reader, FragmentedTrajectoryReader):
+            length = reader._lengths[itraj]
         else:
+            # TODO: Do we really need this? Perhaps the other cases can be collapsed into the first if-branch.
```

Comment:
On a second thought, the proper way to do this would be to overload reader.trajectory_length for the fragmented reader.